doc: release notes: Add Bluetooth relnotes for 2.7.0 (#39112)
zephyrproject-rtos/zephyr: doc/releases/release-notes-2.7.rst

Bluetooth
*********

* Updated all APIs and internal implementation to be conformant with the new
  inclusive terminology in version 5.3 of the Bluetooth Core Specification

* Audio

  * Added the Microphone Input Control Service and client.
  * Split the configuration options into connected and broadcast groups
  * Added support for a new sent callback to be notified when an SDU has been

>     transmitted

Reviewer: It's a good question whether this should be under "Audio", under "Host" or in its own separate "Isochronous" grouping, as these two changes were for ISO, and ISO != Audio, but merely a feature used by LE Audio.

Reviewer: As an additional comment for ISO, I think this misses the rather significant change to the ISO API (as the text here only mentions configurations options). 

I'd add something like 
` * Change connected isochronous API to use group-based opaque struct`

Author: >It's a good question whether this should be under "Audio", under "Host" or in its own separate "Isochronous" grouping, as these two changes were for ISO, and ISO != Audio, but merely a feature used by LE Audio.

For the sake of simplicity in the release notes, let's group those under Audio, since there are no other users of ISO outside of audio today?

Author: >As an additional comment for ISO, I think this misses the rather significant change to the ISO API (as the text here only mentions configurations options).

Added this now

* Direction Finding

  * Added configurability for conditional CTE RX support
  * Added support for CTE periodic advertising chain transmissions

>   * Added support for connection-oriented CTE request and response

Reviewer 2: Is this available upstream? isnt this using new LLCP work?

Reviewer 3: No it is not, please remove it. That is implemented in about 30%...

* Host

  * Added support for setting more than 251 bytes of advertising data
  * Added new callbacks on ATT MTU update
  * Added a new call to retrieve the handle of an advertising set
  * Fixed key overwrite algorithm when working with multiple connections
  * Added configuration support for GATT security re-establishment
  * Added support for writing a long device name
  * OTS: Added object name write capability

* Mesh

  * Added return value for opcode callback
  * Added support for OOB Public Key in the provisionee role
  * Added a new API to manually store pending RPL entries
  * Added support for application access to mesh messages
  * Refactored the Mesh Model Extensions

* Bluetooth LE split software Controller

>   * Added support for advertising PDU chaining, implementing advertising trains

Reviewer 2: ```suggestion
  * Added support for advertising PDU chaining, implementing advertising trains for Direction Finding
```

>   * Added support for adding or removing the ACAD fiel in BIG

Reviewer 2: ```suggestion
  * Added support for adding or removing the ACAD field in Common Extended Header Format to support BIGInfo
```

  * Refactored the legacy, extended and periodic advertising time reservation
    slot calculations

>   * Implemented CSA#2 in connection sub-events

Reviewer 2: ```suggestion
  * Implemented CSA#2 in Extended Advertising and Broadcast ISO sub-events
```
